registration/static/js/src/registration.js

```diff
@@ -54,10 +53,45 @@ app.controller('Registration', ['$scope', '$http', 'djangoForm', function($scope
         return getFormFields().filter(key => $scope.form[key].$dirty || $scope.form[key].$message);
     };
 
+    // Returns the result of validating password using zxcvbn.
+    var validatePassword = function() {
+        var password = $scope.form.password.$viewValue;
+        return zxcvbn(password);
+    };
+
+    // Makes sure each suggestion ends with period.
+    var formatSuggestions = function(suggestions) {
+        return _.map(suggestions, function(suggestion) {
+            if (!suggestion.endsWith('.')) {
+                return suggestion + '.';
+            }
+            return suggestion;
+        });
+    };
+
+    // Returns feedback from password validation.
+    var getValidationFeedback = function() {
+        var validationResult = validatePassword(),
+            suggestions = validationResult.feedback.suggestions,
+            warning = validationResult.feedback.warning;
+        if (warning) {
+            suggestions.push(warning);
+        }
+        suggestions = formatSuggestions(suggestions);
+        // Concatenate suggestions to ensure that all of them are displayed
+        return [suggestions.join(" ")];
```

Comment:
@omarkhan Yes, it looks like this might be a django-angular bug. I'm not too familiar with it but I found that only a single message/suggestion was displayed when `errors` contained multiple messages for a given form field.